sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1960,6 +1987,42 @@ class Dataset[T] private[sql](
       toDF(), colNames.map(colName => resolve(colName)), RelationalGroupedDataset.CubeType)
   }
 
+  /**
+   * Create multi-dimensional aggregation for the current Dataset using the specified grouping sets,
+   * so we can run aggregation on them.
+   * See [[RelationalGroupedDataset]] for all the available aggregate functions.
+   *
+   * This is a variant of groupingSets that can only group by existing columns using column names
+   * (i.e. cannot construct expressions).
+   *
+   * {{{
+   *   // Compute the average for all numeric columns group by specific grouping sets.
+   *   ds.groupingSets(Seq(Seq("department", "group"),Seq()),"department", "group").avg()
+   *
+   *   // Compute the max age and average salary, group by specific grouping sets.
+   *   ds.groupingSets(Seq($"department", $"gender"), Seq($"department"), Seq()).agg(Map(
+   *     "salary" -> "avg",
+   *     "age" -> "max"
+   *   ))
+   * }}}
+   *
+   * @group untypedrel
+   * @since 4.0.0
+   */
+  @scala.annotation.varargs
+  def groupingSets(
+                    groupingSets: Seq[Seq[String]],
```

Comment:
Shall we stop adding string column overloads for new APIs? also cc @HyukjinKwon

Reply:
Are you saying that the methods using string columns are no longer needed, and that I only need to keep the methods using columns of type Column?

Reply:
Yes

Reply:
Got it. This API has been removed.